File: src/main/java/guru/xgm/language/angelscript/lang/folding/AngelScriptCodeFoldingOptionsProvider.java
package guru.xgm.language.angelscript.lang.folding;

import com.intellij.application.options.editor.CodeFoldingOptionsProvider;
import com.intellij.openapi.options.BeanConfigurable;

public class AngelScriptCodeFoldingOptionsProvider extends BeanConfigurable<AngelScriptCodeFoldingSettings> implements CodeFoldingOptionsProvider {
    public AngelScriptCodeFoldingOptionsProvider() {
        super(AngelScriptCodeFoldingSettings.getInstance(), "AngelScript");
        final var settings = getInstance();
        assert settings != null;
        checkBox("Enum", settings::isFoldEnum, settings::setFoldEnum);
    }
}

File: src/main/kotlin/raft/war/language/angelscript/lang/folding/AngelScriptCodeFoldingOptionsProvider.kt
package raft.war.language.angelscript.lang.folding

import com.intellij.application.options.editor.CodeFoldingOptionsProvider
import com.intellij.openapi.options.BeanConfigurable

class AngelScriptCodeFoldingOptionsProvider :
    BeanConfigurable<AngelScriptCodeFoldingSettings?>(AngelScriptCodeFoldingSettings.instance, "AngelScript"),
    CodeFoldingOptionsProvider {
    init {
        val settings = instance
        checkBox("Enum", { settings.isFoldEnum }, { foldEnum: Boolean? -> settings.isFoldEnum = foldEnum!! })
    }
}
